libs/auth-api-lib/src/lib/delegations/delegations-index.service.ts

```diff
@@ -99,27 +116,148 @@ export class DelegationsIndexService {
     )
   }
 
+  /* Index incoming custom delegations */
+  async indexCustomDelegations(nationalId: string) {
+    const delegations = await this.getCustomDelegations(nationalId)
+    await this.saveToIndex(delegations)
+  }
+
+  /* Index incoming personal representative delegations */
+  async indexRepresentativeDelegations(nationalId: string) {
+    const delegations = await this.getRepresentativeDelegations(nationalId)
+    await this.saveToIndex(delegations)
+  }
+
   /*
    * Private methods
    * */
-  private deleteDelegationIndexItems(
-    nationalId: string,
-    delegationTypesToDelete: DelegationType[] = [
-      DelegationType.Custom,
-      DelegationType.LegalGuardian,
-      DelegationType.ProcurationHolder,
-      DelegationType.PersonalRepresentative,
-    ],
-  ) {
-    return Promise.all(
-      delegationTypesToDelete.map((type) =>
-        this.delegationIndexModel.destroy({
+  private async saveToIndex({ created, updated, deleted }: SortedDelegations) {
+    await Promise.all([
+      this.delegationIndexModel.bulkCreate(created),
+      updated.map((d) =>
+        this.delegationIndexModel.update(d, {
           where: {
-            toNationalId: nationalId,
-            type,
+            fromNationalId: d.fromNationalId,
+            toNationalId: d.toNationalId,
+            provider: d.provider,
+            type: d.type,
           },
         }),
       ),
+      this.delegationIndexModel.destroy({
+        where: {
+          fromNationalId: deleted.map((d) => d.fromNationalId),
+          toNationalId: deleted.map((d) => d.toNationalId),
+          provider: deleted.map((d) => d.provider),
+          type: deleted.map((d) => d.type),
```

Comment:
Should this not be like above for updated? 🤔 

Will this work as intended? Trying to figure it out in my head.

Let's say deleted contains:

```typescript
const deleted = [
  { fromNationalId: '1001', toNationalId: '1002', /* ... */ },
  { fromNationalId: '1005', toNationalId: '1007', /* ... */ },
  { fromNationalId: '1010', toNationalId: '1012', /* ... */ },
  { fromNationalId: '1020', toNationalId: '1030', /* ... */ },
]
```

Will this destroy query all delegationIndices where any of this condition applies:
* fromNationalId is in ['1001', '1005', '1010', '1020']
* toNationalId is in ['1002', '1007', '1012', '1030']

So if there are indicies with these same ids that are not supposed to get deleted, won't there be a risk that they will? Maybe I'm wrong, but it feels weird.